### PR TITLE
properly support job host summary data in custom notification templates (and remove a few fields that don't work)

### PR DESCRIPTION
**Pull request by ryanpetrello**

### CODE REVIEW

**Comment by ryanpetrello** [comment]
image

image

**Comment by gamuniz** [comment]
you think you could extend this to also include the other options that aren't being populated:

{'playbook_counts': ['play_count', 'task_count']},

{'project_update': ['id', 'name', 'description', 'status', 'failed']}

{'source_workflow_job': ['description', 'elapsed', 'failed', 'id', 'name', 'status']}

**Comment by ryanpetrello** [comment]
@gamuniz I've removed those, because they don't work.

@jladdjr what's the process for getting @tvo318 what she needs to correct these fields I removed?

File: awx/main/models/notifications.py [hunk 1]
```diff
@@ -270,21 +270,19 @@ class JobNotificationMixin(object):
                             'elapsed', 'job_explanation', 'execution_node', 'controller_node', 'allow_simultaneous',
                             'scm_revision', 'diff_mode', 'job_slice_number', 'job_slice_count', 'custom_virtualenv',
                             'approval_status', 'approval_node_name', 'workflow_url',
-                            {'host_status_counts': ['skipped', 'ok', 'changed', 'failures', 'dark']},
-                            {'playbook_counts': ['play_count', 'task_count']},
```
Comment (ryanpetrello) [inline awx/main/models/notifications.py:274]: This doesn't work (and from what I can tell, never did).

File: awx/main/models/notifications.py [hunk 2]
```diff
@@ -270,21 +270,19 @@ class JobNotificationMixin(object):
...
+                            {'host_status_counts': ['skipped', 'ok', 'changed', 'failed', 'failures', 'dark'
+                                                    'processed', 'rescued', 'ignored']},
                             {'summary_fields': [{'inventory': ['id', 'name', 'description', 'has_active_failures',
                                                                'total_hosts', 'hosts_with_active_failures', 'total_groups',
                                                                'has_inventory_sources',
                                                                'total_inventory_sources', 'inventory_sources_with_failures',
                                                                'organization_id', 'kind']},
                                                 {'project': ['id', 'name', 'description', 'status', 'scm_type']},
-                                                {'project_update': ['id', 'name', 'description', 'status', 'failed']},
```
Comment (ryanpetrello) [inline awx/main/models/notifications.py:281]: This doesn't work (and from what I can tell, never did).

File: awx/main/models/notifications.py [hunk 3]
```diff
@@ -270,21 +270,19 @@ class JobNotificationMixin(object):
...
                                                 {'job_template': ['id', 'name', 'description']},
                                                 {'unified_job_template': ['id', 'name', 'description', 'unified_job_type']},
                                                 {'instance_group': ['name', 'id']},
                                                 {'created_by': ['id', 'username', 'first_name', 'last_name']},
-                                                {'labels': ['count', 'results']},
-                                                {'source_workflow_job': ['description', 'elapsed', 'failed', 'id', 'name', 'status']}]}]
```
Comment (ryanpetrello) [inline awx/main/models/notifications.py:287]: This doesn't work (and from what I can tell, never did).

**Comment by jladdjr** [comment]
> @jladdjr what's the process for getting @tvo318 what she needs to correct these fields I removed?

@ryanpetrello - I think the two relevant docs that would need to be updated are:

* https://github.com/ansible/product-docs/blob/master/userguide/source/notifications.rst#create-custom-notifications
* https://github.com/ansible/product-docs/blob/master/installandreference/source/notification_parameters_supported.rst

The latter page has a detailed break-down of the fields that can be used in custom notifications.

**Review by jladdjr — APPROVED** [review]

**Comment by ryanpetrello** [comment]
https://github.com/ansible/product-docs/pull/1021

**Pull request merged**